dll/ntdll/ldr/ldrinit.c

```diff
@@ -2365,7 +2365,7 @@ LdrpInitializeProcess(IN PCONTEXT Context,
     {
         WCHAR szCSDVersion[128];
         LONG i;
-        ULONG Length = ARRAYSIZE(szCSDVersion) - 1;
+        USHORT Length = (USHORT)ARRAYSIZE(szCSDVersion) - 1;
```

Comment:
Why not instead keeping the ULONG or using SIZE_T instead, and then checking for maximum length and later casting only to USHORT when `Peb->CSDVersion.Length` and `Peb->CSDVersion.MaximumLength` are assigned using this variable? (as is done in LdrpAllocateUnicodeString(), see your changes in ldrutils.c)

Reply:
Because it doesn't make sense to check the size of an array, which we know at compile time

Reply:
Ah yes, good point.